### DISCUSSION TITLE
Dynamic Footer

### DISCUSSION THREAD

**Discussion by cimd** [quasarframework/quasar#17065] [General - Components / Directives / etc]
Hi,

Is there a way to dynamically define a component for the footer, depending on the route visited? 
Example: a "/posts" route would have a different footer from a "/users" route?

**Comment by Dtsiantaris** [comment]
If you mean using the `q-footer` (or even a fixed div, whatever you plan on using), you can use dynamically applied styles/classes. A clean way would be something like this:

```
import { CSSProperties } from "vue"

const FOOTER_STYLES : Record<string, CSSProperties> = {
    "/posts" : {
          background color: red;
    }
}
```
Then in the template you use it like this:

```
<q-footer :style="FOOTER_STYLES[$route.path]>
//CONTENT HERE
</q-footer>
```

**Reply by cimd** [reply]
Thanks, but I didn't mean different styles. I meant completely different components would be imported depending on the route. Eg: one route would have a button, another route would have an image.

**Reply by Dtsiantaris** [reply]
Damn my bad for that. 
What I would propose for your actual problem, if you have many different cases of a footer depending on the route, is to create a `CustomWrapperFooter`. 
Then inside there you could use `v-if`'s to conditionally render the appropriate footer.

```
// Inside the CustomWrapperFooter
<PostsPageFooter v-if="$route.path="/posts" />
//Other possible footers here
// Generic footer that you will reuse in many routes
<GenericFooter v-else />
```

You can still use what I described for styling. Or if you're brave enough you can use something like this:
```
Import { defineComponent } from "vue"
import { PostsFooter } from "src/components"

const FOOTER_COMPONENTS : Record<string, ReturnType<typeof defineComponent>> = {
    "/posts" : PostsComponent
}
```
So in the template I would do something like this:
```
<component :is="FOOTER_COMPONENTS[$route.path] />
```

EDIT: There is surely a better way to make this more Typescript friendly(talking about the Record typing) but you get the picture.

**Reply by RamonDonadeu** [reply]
Actually I Would say you should use layouts.

https://quasar.dev/quasar-cli-webpack/directory-structure/
https://quasar.dev/layout/layout

Here you create a Layout with a given structure

Eg. MainLayout
```
<template>
  <q-layout view="hHh lpR fFf">
    <header />
    <router-view />
    <footer />
  </q-layout>
</template>
```

And then another one with different structure.
Eg. MainLayout
```
<template>
  <q-layout view="hHh lpR fFf">
    <header />
    <router-view />
    <secondFooter />
  </q-layout>
</template>
```

This way you will get the layout you need and don't need to relay on "ifs"
